app/services/return-logs/setup/submit-readings.service.js

```diff
@@ -0,0 +1,136 @@
+ */
+
+const GeneralLib = require('../../../lib/general.lib.js')
+const ReadingsPresenter = require('../../../presenters/return-logs/setup/readings.presenter.js')
+const ReadingsValidator = require('../../../validators/return-logs/setup/readings.validator.js')
+const SessionModel = require('../../../models/session.model.js')
+
+/**
+ * Orchestrates validating the data for `/return-logs/setup/{sessionId}/readings/{yearMonth}` page
+ *
+ * @param {string} sessionId - The UUID of the current session
+ * @param {object} payload - The submitted form data
+ * @param {object} yar - The Hapi `request.yar` session manager passed on by the controller
+ * @param {string} yearMonth - The year and zero-indexed month to view, eg. `2014-0` for January 2014
+ *
+ * @returns {Promise<object>} If no errors it returns an empty object else the page data for the readings page including
+ * the validation error details
+ */
+async function go(sessionId, payload, yar, yearMonth) {
+  const session = await SessionModel.query().findById(sessionId)
+
+  const [requestedYear, requestedMonth] = _determineRequestedYearAndMonth(yearMonth)
+
+  const validationResult = _validate(payload, session, requestedYear, requestedMonth)
+
+  _addResultToSession(payload, session, requestedYear, requestedMonth, validationResult)
+
+  if (!validationResult) {
+    await session.$update()
+
+    GeneralLib.flashNotification(yar, 'Updated', 'Readings have been updated')
+
+    return {}
+  }
+
+  const formattedData = ReadingsPresenter.go(session, yearMonth)
+
+  return {
+    activeNavBar: 'search',
+    error: validationResult,
+    ...formattedData
+  }
+}
+
+function _addResultToSession(payload, session, requestedYear, requestedMonth, validationResult) {
+  // Extract the lines from the session data for the selected year and month
+  const requestedMonthLines = session.lines.filter((line) => {
+    const endDate = new Date(line.endDate)
+
+    return endDate.getFullYear() === requestedYear && endDate.getMonth() === requestedMonth
+  })
+
+  _updateRequestedMonthLines(payload, requestedMonthLines, validationResult)
+
+  // We then update the session lines with the readings derived from the payload and add any errors
+  requestedMonthLines.forEach((requestedMonthLine) => {
+    const matchedSessionLine = session.lines.find((line) => {
+      return line.endDate === requestedMonthLine.endDate
+    })
+    matchedSessionLine.reading = requestedMonthLine.reading
+
+    if (requestedMonthLine.error) {
+      matchedSessionLine.error = requestedMonthLine.error
+    }
+  })
+}
+
+function _determineRequestedYearAndMonth(yearMonth) {
+  // Splitting a string like `2014-0` by the dash gives us an array of strings ['2014', '0']. We chain `.map(Number)` to
+  // then create a new array, applying the Number() function to each one. The result is an array of numbers [2014, 0].
+  return yearMonth.split('-').map(Number)
+}
+
+function _validate(payload, session, requestedYear, requestedMonth) {
```

Comment:
Few functions in here that need to be reordered in alphabetical order (I know you love that ;) )